lib/client.js

```diff
@@ -246,20 +282,47 @@ class ServiceClient {
             accept: 'application/json'
         }, params.headers);
 
-        return new Promise((resolve, reject) => this.breaker.run(
-            (success, failure) => {
+        const {
+            retries,
+            factor,
+            minTimeout,
+            maxTimeout,
+            randomize,
+            transientErrorCheck,
+            onRetry
+        } = this.options.retryOptions;
+
+        const opts = {
+            retries,
+            factor,
+            minTimeout,
+            maxTimeout,
+            randomize
+        };
+
+        const operation = retry.operation(opts);
+
+        return new Promise((resolve, reject) => operation.attempt((currentAttempt) => {
+            this.breaker.run((success, failure) => {
                 return requestWithFilters(params, this.options.filters)
                     .then(result => {
-                        success(); resolve(result);
+                        success();
+                        resolve(result);
                     })
                     .catch(error => {
-                        failure(); reject(error);
+                        failure();
+                        const shouldRetry = operation.retry(error);
```

Comment:
Ohh, by the way, this is already triggering retry! This should come after the transient error check.

Reply:
Maybe something like this?
```
                        const gonnaRetry = currentAttempt > retries;
                        if (!shouldRetry(error) || !gonnaRetry) {
                            reject(error);
                            return;
                        }
                        if(operation.retry(error)) {
                            onRetry(currentAttempt, error);
                        } 
```

Reply:
Or:
```
const canRetry = currentAttempt > retries;
if (!shouldRetry(error) || !canRetry) {
    reject(error);
    return;
}
if(!operation.retry(error)) {
    reject(error);
    return;
}
onRetry(currentAttempt, error);
```

Reply:
Yeah, the second looks good to me.